examples/todomvc/src/element.rs

```diff
@@ -0,0 +1,235 @@
+                        let tg_el = document;
+
+                        let cb = Closure::wrap(Box::new(move |event: web_sys::Event| {
+                            dbg("got fn call delegated");
+                            if let Some(target_element) = event.target() {
+                                let dyn_target_el: Option<
+                                    &web_sys::Node,
+                                > = wasm_bindgen::JsCast::dyn_ref(&target_element);
+                                if let Some(target_element) = dyn_target_el {
+                                    if let Ok(potential_elements) =
+                                        tg_el.query_selector_all(selector)
+                                    {
+                                        let mut has_match = false;
+                                        /*
+                                        dbg(format!(
+                                            "len: {} {}",
+                                            potential_elements.length(),
+                                            selector
+                                        ).as_str());
+*/
+                                        for i in 0..potential_elements.length() {
+                                            if let Some(el) = potential_elements.get(i) {
+                                                if target_element.is_equal_node(Some(&el)) {
+                                                    has_match = true;
+                                                }
+                                                break;
+                                            }
+                                        }
+
+                                        if has_match {
+                                            handler(event);
+                                        }
+                                    }
+                                }
+                            }
+                        }) as Box<FnMut(_)>);
+
+                        dyn_el.add_event_listener_with_callback_and_bool(
+                            //element_et.add_event_listener_with_callback_and_bool(
+                            event,
+                            cb.as_ref().unchecked_ref(),
+                            use_capture,
+                        );
+                        cb.forget(); // TODO cycle collect
+                    }
+                }
+            }
+            self.el = Some(el);
+        }
+    }
+
+    /// Find child `Element`s from this node
+    pub fn qs_from(&mut self, selector: &str) -> Option<Element> {
+        let mut found_el = None;
+        if let Some(el) = self.el.take() {
+            found_el = Some(Element {
+                el: el.query_selector(selector).ok()?,
+            });
+            self.el = Some(el);
+        }
+        found_el
+    }
+
+    /// Sets the inner HTML of the `self.el` element
+    pub fn set_inner_html(&mut self, value: String) {
```

Comment:
This technically only requires `&self` as it's all interior mutability down to JS, but I could understand this as wanting to start to impose Rust stylistic idioms here too

Reply:
Yeah I was hoping to make Element require borrowing and ownership, especially given we don't have any JS changing the objects. However adding `#[derive(Clone)]` goes against that tbh.